core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -328,6 +328,46 @@ object SparkSubmit {
       }
     }
 
+    // In yarn mode for a python app, add pyspark archives to files
+    // that can be distributed with the job
+    if (args.isPython && clusterManager == YARN) {
+      var pyArchives: String = null
+      if (sys.env.contains("PYSPARK_ARCHIVES_PATH")) {
+        pyArchives = sys.env.get("PYSPARK_ARCHIVES_PATH").get
```

Comment:
Actually that's not true, `getOrElse` takes a closure:

    def getOrElse[B >: A](default: ⇒ B): B 

Which is why you can do `getOrElse(throw SomeException())` (look for it in Spark's code base).